functions/go/set-labels/transformer.go

```diff
@@ -0,0 +1,301 @@
+package main
+
+import (
+	"sort"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+type FieldPath []string
+var (
+	selectorPath = FieldPath{"selector", "matchLabels"}
+	metaLabelsPath = FieldPath{"metadata", "labels"}
+	podSelectorPath = FieldPath{"podSelector", "matchLabels"}
+
+)
+
+
+var _ fn.Runner = &SetLabels{}
+
+// SetLabels supports the set-labels workflow, it uses Config to parse functionConfig, Transform to change the labels
+type SetLabels struct {
+	// labels is the desired labels
+	Labels map[string]string `json:"labels,omitempty"`
+	count int
+}
+
+// Transform updates the labels in the right path using GVK filter and other configurable fields
+func (p *SetLabels) Run(_ *fn.Context, fnConfig *fn.KubeObject, items fn.KubeObjects, results *fn.Results) bool {
+	objects := items.WhereNot(fn.IsLocalConfig)
+	if objects.Len() == 0 || objects[0] == nil {
+		results.Infof("no input resources")
+		return true
+	}
+	// A workaround since kpt creates a FunctionConfig placeholder if users don't provide the functionConfig.
+	if fn.EmptyFunctionConfig(fnConfig) {
+		return false
+	}
+	if len(p.Labels) == 0 {
+		results.Warningf("no `labels` arguments are given in FunctionConfig")
+		return true
+	}
+	p.count = 0
+	for _, o := range objects {
+		err := func() error {
```

Comment:
consider renaming outside `err` from the ones inside this inner function. The shadowing of `err` variable may cause strange bugs in the future.

Reply:
That's a great point!